### PR TITLE
[IMP] survey: add user results print

### PR DESCRIPTION
**Pull request by amdi-odoo**
Add the possibility for users to print their results page.
Also improve the display by replacing the "You can review your answers" clickable text by a button.

Task-3884612

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/survey/views/survey_templates_management.xml [hunk 1]
```diff
@@ -110,7 +110,7 @@
     </template>
 
     <template id="survey_button_retake" name="Survey: retake button">
-        <div>
+        <div class="d-print-none">
             <t t-if="not answer.scoring_success and not answer.is_session_answer">
```
Comment (reth-odoo) [inline addons/survey/views/survey_templates_management.xml:114]: Would put the t-if inside the div, currently it creates an empty div so the `gap-3` class is applied and you get a weird space if "take gain" doesn't appear.

**Review by reth-odoo — COMMENTED** [review]
Works for me, just a tiny layout issue. Feel free to green the task once addressed :)

File: addons/survey/views/survey_templates_print.xml [hunk 2]
```diff
@@ -8,31 +8,36 @@
...
             <div class="wrap">
                 <div class="o_survey_print o_container_small">
-                    <div class='py-5 mt32'>
+                    <div class='pt-5 mt32'>
                         <h1><span t-field='survey.title' class="text-break"/></h1>
                         <t t-if="survey.description"><div t-field='survey.description' class="oe_no_empty text-break"/></t>
-                        <t t-if="review" t-call="survey.survey_button_retake"/>
+                        <div class="d-flex gap-2">
+                            <t t-if="review" t-call="survey.survey_button_retake"/>
+                            <p><a role="button" class="btn btn-secondary btn-lg d-print-none o_survey_user_results_print">
```
Comment (awa-odoo) [inline addons/survey/views/survey_templates_print.xml:16]: So there is no text on this one? Maybe add a title="Print Results"?

**Review by awa-odoo — COMMENTED** [review]
Hello @amdi-odoo 

Some minor remarks, but globally LGTM.

Thanks & cheers!

**Comment by awa-odoo** [comment]
@robodoo r+

**Pull request closed without merging**